shared/folders/render.desktop.js

```diff
@@ -0,0 +1,143 @@
+// @flow
+import React, {Component} from 'react'
+import type {Props} from './render'
+import {Box, Text, Icon} from '../common-adapters'
+import Row from './row'
+import {globalStyles, globalColors} from '../styles/style-guide'
+
+type State = {
+  showIgnored: boolean
+}
+
+const rowKey = users => users && users.map(u => u.username).join('-')
+
+const Ignored = ({showIgnored, ignored, styles, onToggle, isPublic}) => {
+  return (
+    <Box style={stylesIgnoreContainer}>
+      <Box style={styles.topBox} onClick={onToggle}>
+        <Text type='BodySmallSemibold' style={stylesDividerText}>Ignored folders</Text>
+        <Icon type={showIgnored ? styles.iconCaretDown : styles.iconCaretRight} style={stylesIgnoreCaret} />
+      </Box>
+      {showIgnored && <Box style={styles.bottomBox}>
+        <Text type='BodySmallSemibold' style={stylesDividerBodyText}>Ignored folders won't show up on your computer and you won't receive alerts about them.</Text>
+      </Box>}
+      {showIgnored && ignored.map((i, idx) => (
+        <Row
+          key={rowKey(i.users)}
+          {...i}
+          users={i.users}
+          isPublic={isPublic}
+          ignored
+          isFirst={!idx} />
+        ))}
+    </Box>
+  )
+}
+
+class Render extends Component<void, Props, State> {
+  state: State;
+
+  constructor (props: Props) {
+    super(props)
+
+    this.state = {
+      showIgnored: false
+    }
+  }
+
+  render () {
+    const realCSS = `
+      .folder-row .folder-row-hover-action { visibility: hidden }
+      .folder-row:hover .folder-row-hover-action { visibility: visible }
+    `
+
+    const styles = this.props.isPublic ? stylesPublic : stylesPrivate
```

Comment:
since we have private/public styles we divide it here and pass it into sub dumb components